internal/network/network.go

```diff
@@ -0,0 +1,169 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/netip"
+)
+
+var (
+	ErrInterfaceNotFound = errors.New("network interface not found")
+)
+
+type Socket interface {
+	Family() Family
+
+	Type() Socktype
+
+	Fd() int
+
+	Close() error
+
+	Bind(addr Sockaddr) error
+
+	Listen(backlog int) error
+
+	Connect(addr Sockaddr) error
+
+	Accept() (Socket, Sockaddr, error)
+
+	Name() (Sockaddr, error)
+
+	Peer() (Sockaddr, error)
+
+	RecvFrom(iovs [][]byte, flags int) (n, rflags int, addr Sockaddr, err error)
+
+	SendTo(iovs [][]byte, addr Sockaddr, flags int) (int, error)
+
+	Shutdown(how int) error
+
+	GetOptInt(level, name int) (int, error)
+
+	GetOptString(level, name int) (string, error)
+
+	SetOptInt(level, name, value int) error
+
+	SetOptString(level, name int, value string) error
```

Comment:
There a lot of different option value types out there. Do you think it's worth taking the same approach as https://github.com/stealthrocket/wasi-go/pull/45? or should we worry about that if/when necessary?

Reply:
I'll try to stretch this approach as far as I can until it becomes unpractical; then we'll do polymorphism or a union type 👍 

I did a quick review of the options we support in wasi-go, and it's all int/string for now (except BINDTODEVICE, but we don't use or support it -- e.g. there are no tests).